mozilla_bitbar_devicepool/lambdatest/user_script/files/script.py

```diff
@@ -0,0 +1,373 @@
+    # power tests, the framework will have already called adb tcpip
+    # 5555 on the device before it disconnected usb. There should be
+    # no devices connected and we will need to perform an adb connect
+    # to connect to the device. DEVICE_SERIAL will be set to either
+    # the device's serial number or its ipaddress:5555 by the framework.
+    try:
+        adbhost = ADBHost(verbose=True)
+        if env["DEVICE_SERIAL"].endswith(":5555"):
+            # Power testing with adb over wifi.
+            adbhost.command_output(["connect", env["DEVICE_SERIAL"]])
+        devices = adbhost.devices()
+        print(json.dumps(devices, indent=4))
+        if len(devices) != 1:
+            fatal(
+                "Must have exactly one connected device. {} found.".format(
+                    len(devices)
+                ),
+                retry=True,
+            )
+    except (ADBError, ADBTimeoutError) as e:
+        fatal("{} Unable to obtain attached devices".format(e), retry=True)
+
+    try:
+        for f in glob("/tmp/adb.*.log"):
+            print("\n{}:\n".format(f))
+            with open(f) as afile:
+                print(afile.read())
+    except Exception as e:
+        print("{} while reading adb logs".format(e))
+
+    print("Connecting to Android device {}".format(env["DEVICE_SERIAL"]))
+    try:
+        device = ADBDevice(device=env["DEVICE_SERIAL"])
+        android_version = device.get_prop("ro.build.version.release")
+        print(
+            "Android device version (ro.build.version.release):  {}".format(
+                android_version
+            )
+        )
+
+        # power testing
+        #   - 10/23 (aje): disabled as per Sparky we're not going to be doing power testing
+        #     - they will let us know if this changes
+        #
+        # this can explode if an unknown device, explode now vs in an hour...
+        # device_type = get_device_type(device)
+
+        # set device to UTC
+        if device.is_rooted:
+            device.shell_output(
+                'setprop persist.sys.timezone "UTC"', timeout=ADB_COMMAND_TIMEOUT
+            )
+        # show date for visual confirmation
+        device_datetime = device.shell_output("date", timeout=ADB_COMMAND_TIMEOUT)
+        print("Android device datetime:  {}".format(device_datetime))
+
+        # clean up the device.
+        device.rm("/data/local/tests", recursive=True, force=True)
+        device.rm("/data/local/tmp/*", recursive=True, force=True)
+        device.rm("/data/local/tmp/xpcb", recursive=True, force=True)
+        device.rm("/sdcard/tests", recursive=True, force=True)
+        device.rm("/sdcard/raptor-profile", recursive=True, force=True)
+    except (ADBError, ADBTimeoutError) as e:
+        fatal("{} attempting to clean up device".format(e), retry=True)
+
+    if taskcluster_debug:
+        env["DEBUG"] = taskcluster_debug
+
+    print("environment = {}".format(json.dumps(env, indent=4)))
```

Comment:
a lot of the "print" statements here are not necessary; was primarily used for debugging while getting things running.  Feel free to remove or keep them

Reply:
Disabled env dumping and netstat running. Will look for some more.